pkg/server/apiutil/apiutil.go

```diff
@@ -30,3 +32,59 @@ func WriteJSONResponse(ctx context.Context, w http.ResponseWriter, code int, pay
 	}
 	_, _ = w.Write(res)
 }
+
+// GetQueryStringVal gets the string value of a Query String parameter, param, from the provided url query string
+// values.
+//
+// Returns a QueryStringVal struct with the value, if the param exists in queryStringVals. If not, the Exists field
+// will be false.
+func GetQueryStringVal(queryStringVals url.Values, param string) QueryStringVal[string] {
+	if queryStringVals.Has(param) {
+		return QueryStringVal[string]{Value: queryStringVals.Get(param), Exists: true}
+	}
+	return QueryStringVal[string]{Exists: false}
+}
+
+// GetIntQueryStringVal gets the int value of a Query String parameter, param, from the provided url query string
+// values.
+//
+// Returns a QueryStringVal struct with the value, if the param exists in queryStringVals and can be cast to an int.
+// If not, the Exists field will be false.
+func GetIntQueryStringVal(queryStringVals url.Values, param string) QueryStringVal[int] {
+	if queryStringVals.Has(param) {
+		queryArgStr := queryStringVals.Get(param)
+		queryArgInt, err := strconv.Atoi(queryArgStr)
+		if err == nil {
+			return QueryStringVal[int]{Value: queryArgInt, Exists: true}
+		}
+	}
+	return QueryStringVal[int]{Exists: false}
+}
+
+// GetIntQueryStringVals gets all int values of a Query String parameter, param, from the provided url query string
+// values.
+//
+// Returns a QueryStringVal struct with the values, if the params exists in queryStringVals and can be cast to an int.
+// If not, the Exists field will be false.
+func GetIntQueryStringVals(queryStringVals url.Values, param string) QueryStringVal[[]int] {
+	if queryStringVals.Has(param) {
+		queryArgStrs := queryStringVals[param]
+		queryArgInts := make([]int, 0, len(queryArgStrs))
+		for _, a := range queryArgStrs {
+
+			i, err := strconv.Atoi(a)
+			if err != nil {
+				return QueryStringVal[[]int]{Exists: false}
+			}
+			queryArgInts = append(queryArgInts, i)
+		}
+
+		return QueryStringVal[[]int]{Value: queryArgInts, Exists: true}
+	}
+	return QueryStringVal[[]int]{Exists: false}
+}
+
+type QueryStringVal[T any] struct {
```

Comment:
Do you feel strongly about adding this struct? I feel that nil or zero-value checks should be done to indicate existence, eg:

```go
if c.tableName == nil {
```
or
```go
if c.tableName == "" {
```